### PR TITLE
Remove deprecated source_permissions

### LINKED ISSUE #315: message : The `source_permissions` parameter is deprecated
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: any
- Ruby:
- Distribution: any
- Module version: 7.4.0

## What are you seeing

We are using file resource with source_permissions attribute here :
https://github.com/voxpupuli/puppet-openvpn/blob/master/manifests/ca.pp#L63

It looks to do the expected job and docs about types does not show deprecation message.

The deprecation message is relevant : 
https://tickets.puppetlabs.com/browse/PUP-9332

## What behaviour did you expect instead

no deprecation message

## Output log
```
$> puppet apply --verbose --detailed-exitcodes /tmp/apply_manifest.pp.3BVJmE
Info: Loading facts
Info: Loading facts
Notice: Compiled catalog for vpnserver in environment production in 0.21 seconds
Warning: The `source_permissions` parameter is deprecated. Explicitly set `owner`, `group`, and `mode`.
(file: /etc/puppetlabs/code/modules/openvpn/manifests/ca.pp, line: 59)
Info: Applying configuration version '1543266749'
```
## Any additional information you'd like to impart

Probably, it should be possible to use https://github.com/puppetlabs/puppetlabs-rsync , instead of file resource.

### PR DESCRIPTION
**Pull request by jkroepke**
Follow up: #312 

Since I hate deprecation warnings and the old PR looks stale just I create a new.

#### Pull Request (PR) description
```
Fix of deprecation warning:
Warning: The source_permissions parameter is deprecated. Explicitly set owner, group, and mode.
(file: .../manifests/ca.pp, line: 127)
```

#### This Pull Request (PR) fixes the following issues
Fixes #315

### CODE REVIEW

File: manifests/ca.pp [hunk 1]
```diff
@@ -62,21 +62,45 @@
...
+    links   => 'follow',
+    ignore  => '*.cnf',
+    owner   => 0,
+    group   => 0,
+    mode    => '0755',
+    source  => "file:${openvpn::easyrsa_source}",
+    require => File["${etc_directory}/openvpn/${name}"],
+  }
+
+  exec { "copy *.cnf files from easyrsa source to ${name}":
```
Comment (bastelfreak) [inline manifests/ca.pp]: I don't think that copying data with an exec resource is a good idea. @alexjfisher any ideas?

Comment (jkroepke) [reply manifests/ca.pp]: I'm open for new ideas. From https://github.com/voxpupuli/puppet-openvpn/issues/315 I grab the idea with puppetlabs-rsync which use a exec resource to copy data with rsync.

The current solution is a nasty hack. Thats all what I know.

Comment (jkroepke) [reply manifests/ca.pp]: What about restore the old behavior

https://github.com/voxpupuli/puppet-openvpn/commit/9f54ab834654dffb494f3997ca709872d2dedf28#diff-b386e10e7f4de7981ca1d52c90456a0dL127

**Comment by ghoneycutt** [comment]
This would be easier to approve and merge if the PR only dealt with removing the `source_permissions` attributes. It seems to be changing functionality without tests, which makes it hard to merge.

**Comment by jkroepke** [comment]
> It seems to be changing functionality without tests, which makes it hard to merge.

Maybe I need some consulting which tests are required here.

I code this PR and test and try out a lot. If the functionally was wrong, the test 

https://github.com/voxpupuli/puppet-openvpn/blob/48acf2427611593d878a409a43fb54b2f4235a2b/spec/acceptance/openvpn_spec.rb#L61

give a red flag.

Example: https://travis-ci.org/voxpupuli/puppet-openvpn/jobs/609636194

**Pull request closed without merging**